### PR TITLE
docs: init documentation website

### LINKED ISSUE #381: Extend Documentation with docusaurus & github-pages
This is an question/discussion about "outsourcing" and extending the documentation into github pages with docusaurus(2)

(Example (from typegoose): https://typegoose.github.io/typegoose/)

CC @nodkz

### PR DESCRIPTION
**Pull request by hasezoey**
The Website is setup like in typegoose, using the file structure `/docs` and `/website`
To be done:
- add content to the quick-start-guide
- change the texts in the root url for the flavor list ("Easy to Use", etc)
- add content to `/docs/api`
- optional: add favicon
- optional: open an discord server
- after pr is merged: setup algolia (search engine, look in the typegoose for an working example)

Notes:
- the workflow is independend of the tests, and only run on branch `master` and only if files are changed in `/docs` or `/website` (on push)
- the website is setup for repository `nodkz/mongodb-memory-server` and not an/the organization
- the website gets deployed into branch `gh-pages` which keeps history

Run (read the readme in `/website`):
```sh
cd website
yarn install
yarn start # accessible under "http://localhost:3000/mongodb-memory-server"
```

## Related Issues

- closes #381

### CODE REVIEW

**Review by nodkz — COMMENTED** [review]
I don't have any logo. But it quite simple to create 😜 
You may take any or create any new one (the pattern is obvious 😅).

logo
logo2

**Review by nodkz — APPROVED** [review]
I don't know what to do with discord. For now I don't see how it can help us. But we can try if you want.

**Comment by hasezoey** [comment]
> I don't know what to do with discord. For now I don't see how it can help us. But we can try if you want.

i suggested it, because i have one for typegoose, and it helped (faster responses and help integrating it)

> You may take any or create a new one (the pattern is obvious sweat_smile).

sorry, im not an logo designer (or lower), so i cannot make one (~~or at least dont want to do such things~~)

File: docs/guides/quick-start-guide.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+id: quick-start-guide
+title: 'Quick Start Guide'
+---
+
+*empty still*
```
Comment (hasezoey) [inline docs/guides/quick-start-guide.md:6]: this still needs to be extended before merge

File: website/docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      links: [
+        {
+          title: 'Community',
+          items: [
+            {
+              label: 'Stack Overflow',
+              href: 'https://stackoverflow.com/questions/tagged/mongodb-memory-server',
+            },
+            //{
+            //  label: 'Discord',
```
Comment (hasezoey) [inline website/docusaurus.config.js:52]: this can be removed, if not discord will be used

File: website/src/pages/index.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import clsx from 'clsx';
+import Layout from '@theme/Layout';
+import Link from '@docusaurus/Link';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import styles from './styles.module.css';
+
+const features = [
```
Comment (hasezoey) [inline website/src/pages/index.js:9]: the texts still need to be changed (im not good at finding the right words for such things)

**Comment by hasezoey** [comment]
converted back to draft, because this is still not finished (and pull requests can be reviewed while being a draft)

**Review by nodkz — APPROVED** [review]
Let's launch an empty site without content.

**Comment by nodkz** [comment]
Of course in hidden mode. Without adding links to it.
For testing CI, for algolia addition.

**Comment by hasezoey** [comment]
> For testing CI, for algolia addition.

i dont know the full requirements of this anymore, but i think *some* documentation needs to exist

PS: i really dont know what to put in these 3 things (`features` array)

**Comment by hasezoey** [comment]
Documentation for the basic things like `Supported Systems`, `MongoMemoryReplSet` & `MongoMemoryServer` are added

also including some non-docs related stuff (things i noticed while going through the code for the documentation)

---

i would say that documentation is still WIP, but this PR could be merged - to be have the fixes and and basic documentation running

**Pull request merged**